### PR TITLE
Complete rshift implementation

### PR DESCRIPTION
**Pull request by HarshitaKalani**

### CODE REVIEW

**Comment by certik** [comment]
This PR overrides the shiftr implementation with an rshift implementation (in tests, instantiation, etc.). Is that intended?

**Comment by HarshitaKalani** [comment]
Yes. While implementing `shiftr`, I didn't know there exists another function `rshift`. And unknowingly I implemented `rshift` instead of `shiftr`. This PR rectifies that.

File: src/libasr/pass/intrinsic_functions.h [hunk 1]
```diff
@@ -924,6 +925,38 @@ namespace Shiftr {
...
+        int64_t val1 = ASR::down_cast<ASR::IntegerConstant_t>(args[0])->m_n;
+        int64_t val2 = ASR::down_cast<ASR::IntegerConstant_t>(args[1])->m_n;
+        int64_t val = val1 >> val2;
+        return make_ConstantWithType(make_IntegerConstant_t, val, t1, loc);
+    }
+
+    static inline ASR::expr_t* instantiate_Rshift(Allocator &al, const Location &loc,
+            SymbolTable *scope, Vec<ASR::ttype_t*>& arg_types, ASR::ttype_t *return_type,
+            Vec<ASR::call_arg_t>& new_args, int64_t /*overload_id*/) {
+        declare_basic_variables("_lcompilers_rshift_" + type_to_str_python(arg_types[0]));
```
Comment (certik) [inline src/libasr/pass/intrinsic_functions.h:941]: This one is not in the runtime.

Comment (certik) [reply src/libasr/pass/intrinsic_functions.h:941]: If it is not used, then I think we can just use an empty string:
```suggestion
        declare_basic_variables("");
```
That way there is no confusion that it calls into the runtime library.

**Review by certik — APPROVED** [review]
This is good, just one minor comment.

File: src/libasr/pass/intrinsic_functions.h [hunk 2]
```diff
@@ -924,6 +925,38 @@ namespace Shiftr {
...
+        int64_t val1 = ASR::down_cast<ASR::IntegerConstant_t>(args[0])->m_n;
+        int64_t val2 = ASR::down_cast<ASR::IntegerConstant_t>(args[1])->m_n;
+        int64_t val = val1 >> val2;
+        return make_ConstantWithType(make_IntegerConstant_t, val, t1, loc);
+    }
+
+    static inline ASR::expr_t* instantiate_Rshift(Allocator &al, const Location &loc,
+            SymbolTable *scope, Vec<ASR::ttype_t*>& arg_types, ASR::ttype_t *return_type,
+            Vec<ASR::call_arg_t>& new_args, int64_t /*overload_id*/) {
+        declare_basic_variables("" + type_to_str_python(arg_types[0]));
```
Comment (certik) [inline src/libasr/pass/intrinsic_functions.h:941]:
```suggestion
        declare_basic_variables("");
```

File: src/libasr/pass/intrinsic_functions.h [hunk 3]
```diff
@@ -904,7 +905,7 @@ namespace Shiftr {
     static inline ASR::expr_t* instantiate_Shiftr(Allocator &al, const Location &loc,
             SymbolTable *scope, Vec<ASR::ttype_t*>& arg_types, ASR::ttype_t *return_type,
             Vec<ASR::call_arg_t>& new_args, int64_t /*overload_id*/) {
-        declare_basic_variables("_lcompilers_sign_" + type_to_str_python(arg_types[0]));
+        declare_basic_variables("" + type_to_str_python(arg_types[0]));
```
Comment (certik) [inline src/libasr/pass/intrinsic_functions.h:908]:
```suggestion
        declare_basic_variables("");
```

**Comment by HarshitaKalani** [comment]
Ready!

**Comment by HarshitaKalani** [comment]
I thought it was ready to be merged. This caused issues in the main, hence I've reverted it. I'm sorry for this.

**Comment by certik** [comment]
Please only merge reviewed PRs. And when you merge, do either a clean commit history, or squash it. Do not first merge master into the PR and then merge into master. If you merge master into a PR, you need to squash it then (or manually rebase to create a clean set of commits).

**Pull request merged**